tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py

```diff
@@ -271,7 +276,14 @@ def _fit(self,
                 with self.model.no_logging():
                     original_path = self.model.path
                     self.model.path = Path(temp_dir)
-                    self.model.fit_one_cycle(params['epochs'], params['lr'], cbs=callbacks)
+
+                    len_val = len(X_val) if X_val is not None else 0
+                    epochs = self._get_epochs_number(len(X) + len_val, params['epochs'], batch_size, time_left)
+                    if epochs == 0:
```

Comment:
what if time_left is negative which results in epochs being negative?

Reply:
good catch! added a test case and handling. It should return `0`.